experimental/framework/fluid-llm/src/test/object-diff/objectDiffArray.spec.ts

```diff
@@ -0,0 +1,247 @@
+import { strict as assert } from "node:assert";
+
+import { objectDiff, type Difference } from "../../object-diff/index.js";
+
+describe("objectDiff - arrays", () => {
+	it("top level array & array diff", () => {
+		assert.deepStrictEqual(objectDiff(["test", "testing"], ["test"]), [
+			{
+				type: "REMOVE",
+				path: [1],
+				oldValue: "testing",
+			},
+		]);
+	});
+
+	it("nested array", () => {
+		assert.deepStrictEqual(objectDiff(["test", ["test"]], ["test", ["test", "test2"]]), [
+			{
+				type: "CREATE",
+				path: [1, 1],
+				value: "test2",
+			},
+		]);
+	});
+
+	it("object in array in object", () => {
+		assert.deepStrictEqual(
+			objectDiff({ test: ["test", { test: true }] }, { test: ["test", { test: false }] }),
+			[
+				{
+					type: "CHANGE",
+					path: ["test", 1, "test"],
+					value: false,
+					oldValue: true,
+				},
+			],
+		);
+	});
+
+	it("array to object", () => {
+		assert.deepStrictEqual(objectDiff({ data: [] }, { data: { val: "test" } }), [
+			{ type: "CHANGE", path: ["data"], value: { val: "test" }, oldValue: [] },
+		]);
+	});
+});
+
+describe("objectDiff - arrays with object ID's", () => {
+	it("object with id is moved from a new deleted array index", () => {
+		const diffs: Difference[] = objectDiff(
+			{ state: ["test", { id: "1", test: true }] },
+			{ state: [{ id: "1", test: true }] },
+			{
+				cyclesFix: true,
+				useObjectIds: {
+					idAttributeName: "id",
+				},
+			},
+		);
+
+		assert.deepStrictEqual(diffs, [
+			{
+				type: "CHANGE",
+				path: ["state", 0],
+				value: { id: "1", test: true },
+				oldValue: "test",
+			},
+			{
+				type: "MOVE",
+				path: ["state", 1],
+				newIndex: 0,
+				value: { id: "1", test: true },
+			},
+		]);
```

Comment:
This one confuses me too. The old state has two items, the end state just has the second one of those. So I would have expected a DELETE operation somewhere. A CHANGE + a MOVE makes me think that the string at the first position would become an object _identical_ to the second one _but not the same object_, and then after a MOVE on top of that I would expect to still have two items in the array, the one originally at index 1 now at index 0, and at index 1 a new one that _looks_ like the one which was originally there.

Reply:
This one is an unhandled case that's occurring because we have an array with multiple different types within it. 

I could modify the code to handle the object "id" strategy within an array that may or may not contain nodes with id's but it probably doesn't make sense. Also, I'm going to remove the object-diff/ folder from this PR as that is the older version of the diffing algorithm (I have the shared-tree-diff/ folder).